BaseTools/Source/Python/AutoGen/GenC.py

```diff
@@ -2039,6 +2042,34 @@ def CreateFooterCode(Info, AutoGenC, AutoGenH):
 def CreateCode(Info, AutoGenC, AutoGenH, StringH, UniGenCFlag, UniGenBinBuffer, StringIdf, IdfGenCFlag, IdfGenBinBuffer):
     CreateHeaderCode(Info, AutoGenC, AutoGenH)
 
+    # The only 32 bit archs we have are IA32 and ARM, everything else is 64 bit
+    Bitwidth = 32 if Info.Arch == 'IA32' or Info.Arch == 'ARM' else 64
+
+    if GlobalData.gStackCookieValues64 == [] and os.path.exists(os.path.join(Info.PlatformInfo.BuildDir, "StackCookieValues64.json")):
+        with open (os.path.join(Info.PlatformInfo.BuildDir, "StackCookieValues64.json"), "r") as file:
+            GlobalData.gStackCookieValues64 = json.load(file)
+    if GlobalData.gStackCookieValues32 == [] and os.path.exists(os.path.join(Info.PlatformInfo.BuildDir, "StackCookieValues32.json")):
+        with open (os.path.join(Info.PlatformInfo.BuildDir, "StackCookieValues32.json"), "r") as file:
+            GlobalData.gStackCookieValues32 = json.load(file)
+
+    try:
+        if Bitwidth == 32:
+            CookieValue = int(GlobalData.gStackCookieValues32[hash(Info.Guid) % len(GlobalData.gStackCookieValues32)])
+        else:
+            CookieValue = int(GlobalData.gStackCookieValues64[hash(Info.Guid) % len(GlobalData.gStackCookieValues64)])
+    except:
+        EdkLogger.warn("build", "Failed to get Stack Cookie Value List! Generating random value.", ExtraData="[%s]" % str(Info))
+        if Bitwidth == 32:
+            CookieValue = secrets.randbelow (0xFFFFFFFF)
+        else:
+            CookieValue = secrets.randbelow (0xFFFFFFFFFFFFFFFF)
+
+    AutoGenH.Append((
+        '#define STACK_COOKIE_VALUE 0x%XULL\n' % CookieValue

```

Comment:
We have requirements for build reproducibility so I think we would need a way to turn this off? We don't want randomness injected into the image.

Reply:
@ajfish , `STACK_COOKIE_VALUE` only gets used in the actual StackCheckLib implementation. The way to turn this off is to use StackCheckLibNull, which doesn't reference this value, so it is not used in the binary.